pkg/gofr/container/health.go

```diff
@@ -54,6 +58,41 @@ func (c *Container) Health(ctx context.Context) interface{} {
 	return healthMap
 }
 
+func checkExternalDBHealth(c *Container, downCount int,
+	healthMap map[string]interface{}) (updatedDownCount int, updatedHealthMap map[string]interface{}) {
+	updatedDownCount = downCount
+	updatedHealthMap = healthMap
+
+	if !isNil(c.Mongo) {
+		health := c.Mongo.HealthCheck()
+		if strings.Contains(fmt.Sprint(health), "DOWN") {
```

Comment:
I'm sorry, but this is too strange.

HealthCheck is returning any, then you use the fmt.Stringer to check if it contains the string down ?

Why not having this, and each implementation returns an error, and a ErrHeathCheckFailed you can catch?

```go
type HealthChecker interface {
	HealthCheck(context.Context) error
}
```

This way you can return a struct that implements Error()


Or, you could return an interface like this

```go
type HealthChecker interface {
	HealthCheck(context.Context) HealthStatus
}

type HealthStatus interface {
    IsDown() bool
    GetMetadata() any
}
```

So you could use GetMetadata to assign the health status of each component.

Please note you can combine both

And do

```go
type HealthChecker interface {
	HealthCheck(context.Context) HealthStatus
}

type HealthStatus interface {
    error
    GetMetadata() any
}
```

 I find too strange to return the whole heathcheck results.

You may choose, one solution, or even think about yours, but using `strings.Contains("DOWN", fmt.Sprint(health))` sounds too strange

Reply:
Would be nice it it was 

```
type HealthStatus interface {
    IsDown() bool
    GetMetadata() any
}
```

with GetMetadata() being lazy (which means not collecting metrics unless specifically asked for it)
This would help a lot while implementing Readiness probe as discussed here https://discord.com/channels/1156248707931590666/1257790439260553298

Reply:
could you invite me to the discord server you are talking about? I'm not in, at Discord is showing me something strange

Reply:
@ccoVeille It's the discord listed on Gofr website
You can use this link: https://discord.gg/5ACeSKGt37
or use the one here
![image](https://github.com/gofr-dev/gofr/assets/20914644/3b013aba-4adf-477a-87c9-2c86f677d9fd)

Reply:
> Would be nice it it was 
> 
> ```
> type HealthStatus interface {
>     IsDown() bool
>     GetMetadata() any
> }
> ```
> 
> with GetMetadata() being lazy (which means not collecting metrics unless specifically asked for it)
> This would help a lot while implementing Readiness probe as discussed here https://discord.com/channels/1156248707931590666/1257790439260553298
>  

I would recommend you using interface

```go
type HealthStatus interface {
     GetMetadata() (any, error)
 }
```

This way you can return a sentinel error when down.

Reply:
https://discord.com/invite/wsaSkQTdgq
Can you try using this link

Reply:
> > Would be nice it it was
> > ```
> > type HealthStatus interface {
> >     IsDown() bool
> >     GetMetadata() any
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > with GetMetadata() being lazy (which means not collecting metrics unless specifically asked for it)
> > This would help a lot while implementing Readiness probe as discussed here https://discord.com/channels/1156248707931590666/1257790439260553298
> 
> I would recommend you using interface
> 
> ```go
> type HealthStatus interface {
>      GetMetadata() (any, error)
>  }
> ```
> 
> This way you can return a sentinel error when down.

but you would need to gather stats every time you have to check for readiness. Which isn't ideal
This is ok thou:
```
type HealthStatus interface {
    IsDown() bool
    GetMetadata() (any, error)
}
```

Reply:
I'm unsure about the exact needs, but anything like this would be better than the current sprint() + contains("down") 😅

Reply:
Since we would any way be fetching stats, it would be better to implement the following.

With the condition such as 
If there is an Error the service in obviously down and the resp in that case is the error details.

If there is no error then it is up with response having the details.
```
type HealthChecker interface {
	HealthCheck() (any, error)
}
```

Reply:
Thanks for seconding my suggestion